lstchain/reco/dl0_to_dl1.py

```diff
@@ -383,24 +384,30 @@ def r0_to_dl1(input_filename = get_dataset_path('gamma_test_large.simtel.gz'),
                                 event.lst.tel[telescope_id].evt.tib_pps_counter +
                                 event.lst.tel[telescope_id].evt.tib_tenMHz_counter * 10**(-7))
 
-                        dl1_container.ucts_time = ucts_time
-                        dl1_container.dragon_time = dragon_time
-                        dl1_container.tib_time = tib_time
+                        # FIXME: directly use unix_tai format whenever astropy v4.1 is out
```

Comment:
Could you make these functions and add a test, that the expected number of seconds are there?

Something like:
```
def unix_tai_to_utc(timestamp):
    '''Workaround for unix time in scale TAI''' 
    tai = Time(timestamp, format='unix', scale='tai')
    return Time(tai.utc.jd, format='jd', scale='tai')
```

Reply:
@MaxNoe, the test function should be placed in `lstchain/reco/tests/test_utils.py` script?

Reply:
If the functions lives in `lstchain/reco/utils.py`, yes

Reply:
Should be the test function something like the following?

```python
def test_unix_tai_to_utc():
    timestamp = 1579376359.3225002
    tai = Time(timestamp, format='unix', scale='tai')
    utc = Time(tai.utc.jd, format='jd', scale='tai')
    np.testing.assert_allclose(tai.unix, utc.unix) 
```

or should explicitly consider the 37 leap seconds somehow?

Reply:
If these two are `allclose`, isn't the conversion doing nothing?

Reply:
Let me understand, what the input timestamp represents. Is this si seconds since `1970-01-01T00:00:00 TAI`?

Reply:
Actually `rtol` was too relaxed and tai and utc were indeed `all_closed`.

This is at least considering those 37 secs:
```python
def test_unix_tai_to_utc():
    timestamp = 1579376359.3225002
    leap_seconds = 37
    tai = Time(timestamp, format='unix', scale='tai')
    utc = Time(tai.utc.jd, format='jd', scale='tai')
    np.testing.assert_allclose(tai.unix, utc.unix, rtol=1e-12, atol=leap_seconds) 
```

Reply:
You should define a function and test that function. A test that tests lines of code that are not used somewhere else, is not really helpful.

So define the function and test if it is what you expect.

Reply:
True... I was not aware of that
It is the first time I implement a test.

Provided the function is:

```python
def unix_tai_to_utc(timestamp):
    '''Workaround for unix time in scale TAI'''
    tai = Time(timestamp, format='unix', scale='tai')
    return Time(tai.utc.jd, format='jd', scale='tai')
```

This could serve as a test?
```python
def test_unix_tai_to_utc():
    timestamp_tai = 1579376359.3225002
    leap_seconds = 37
    utc_time = unix_tai_to_utc(timestamp_tai)
    np.testing.assert_allclose(utc_time.unix, timestamp_tai - leap_seconds, rtol=1e-12)
```

Reply:
Yes, but I still think is is a bit of a convoluted, unintuitive solution. Adding the timestamp to the epoch is much clearer I think.

Reply:
I agree. I implemented it.